internal/controller/ols_app_redis_assets.go

```diff
@@ -0,0 +1,102 @@
+package controller
+
+import (
+	olsv1alpha1 "github.com/openshift/lightspeed-operator/api/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+func (r *OLSConfigReconciler) generateRedisDeployment(cr *olsv1alpha1.OLSConfig) (*appsv1.Deployment, error) {
+	cacheReplicas := int32(1)
+	DeploymentSelectorLabels := map[string]string{
+		"app.kubernetes.io/component":  "redis-server",
+		"app.kubernetes.io/managed-by": "lightspeed-operator",
+		"app.kubernetes.io/name":       "lightspeed-service-redis",
+		"app.kubernetes.io/part-of":    "openshift-lightspeed",
+	}
+	deployment := appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      OLSAppRedisDeploymentName,
+			Namespace: cr.Namespace,
+			Labels:    DeploymentSelectorLabels,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &cacheReplicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: DeploymentSelectorLabels,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: DeploymentSelectorLabels,
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:            "lightspeed-redis-server",
+							Image:           r.Options.LightspeedServiceRedisImage,
+							ImagePullPolicy: corev1.PullAlways,
```

Comment:
maybe this is for testing purpose.
I think it better to pull the image if not present locally.

Reply:
Changed as suggested.